btk/measure.py

```diff
@@ -224,12 +224,21 @@ def __next__(self):
         """
         blend_output = next(self.draw_blend_generator)
         input_args = ((blend_output, i) for i in range(self.batch_size))
-        measure_results = multiprocess(
+        measure_output = multiprocess(
             self.run_batch,
             input_args,
             self.cpus,
             self.verbose,
         )
         if self.verbose:
             print("Measurement performed on batch")
+        measure_results = {}
+        for i, f in enumerate(self.measure_functions):
+            measure_dic = {}
+            for key in ["catalog", "deblended_images", "segmentation"]:
+                if key in measure_output[0][i] and measure_output[0][i][key] is not None:
```

Comment:
I think you only need the second check in this 'and' clause (please let me know if I'm forgetting something though!)

Reply:
I think that if we try to access it and the entry does not exist it will throw an error. But I guess that they should be set to None anyway if they don't exist so you may be right.

Reply:
Yeah I always set them to `None`, I have it in the docstring too: ```Omitted keys in the returned dictionary are automatically assigned a `None` value (except for
`catalog` which is a mandatory entry).
```

Reply:
I think it's best to avoid checks that will never fail?